core/src/main/java/org/apache/iceberg/BaseEntriesTable.java

```diff
@@ -125,31 +130,120 @@ ManifestFile manifest() {
 
     @Override
     public CloseableIterable<StructLike> rows() {
-      // Project data-file fields
-      CloseableIterable<StructLike> prunedRows;
-      if (manifest.content() == ManifestContent.DATA) {
-        prunedRows =
+      Types.NestedField readableMetricsField = projection.findField(MetricsUtil.READABLE_METRICS);
+
+      if (readableMetricsField == null) {
+        CloseableIterable<StructLike> entryAsStruct =
             CloseableIterable.transform(
-                ManifestFiles.read(manifest, io).project(fileSchema).entries(),
-                file -> (GenericManifestEntry<DataFile>) file);
+                entries(fileProjection),
+                entry -> (GenericManifestEntry<? extends ContentFile<?>>) entry);
+
+        StructProjection structProjection = projectNonReadable(projection);
+        return CloseableIterable.transform(entryAsStruct, structProjection::wrap);
       } else {
-        prunedRows =
-            CloseableIterable.transform(
-                ManifestFiles.readDeleteManifest(manifest, io, specsById)
-                    .project(fileSchema)
-                    .entries(),
-                file -> (GenericManifestEntry<DeleteFile>) file);
+        Schema requiredFileProjection = requiredFileProjection();
+        Schema actualProjection = removeReadableMetrics(readableMetricsField);
+        StructProjection structProjection = projectNonReadable(actualProjection);
+
+        return CloseableIterable.transform(
+            entries(requiredFileProjection),
+            entry -> withReadableMetrics(structProjection, entry, readableMetricsField));
       }
+    }
+
+    /**
+     * Remove virtual columns from the file projection and ensure that the underlying metrics used
```

Comment:
Hey I'm sorry I missed this, can we remove this line "Remove virtual columns from the file projection and."  

I guess this comment was rather for the other operation, but I dont think we need the comment anymore now that its on a self explanatory method.